### DISCUSSION TITLE
Will battery module get auto disabled if pc is a desktop ?

### DISCUSSION THREAD

**Discussion by mobsenpai** [Alexays/Waybar#3175] [Q&A] [answered]
I want the battery module to dynamically disable if I am not on a laptop i.e using the same config on desktop and laptop. Will it get auto disabled or maybe I can use some `mkif` function for nixos if anyone has done it before, do tell.

Any help appreciated!

**Comment by Siri-chan** [comment]
It should. Waybar hides the battery module if it can't find the battery in `/sys/class/power_supply/`. You should be able to test this on a laptop, by manually setting the `bat` key to a battery that doesn't exist on your laptop, and starting a second waybar.

For example, on my laptop, changing `bat` from `BAT1` to `BAT2` is enough to hide it.

**Reply by Siri-chan** [reply, marked as answer]
Further clarification: A desktop shouldn't export any batteries to `/sys/class/power_supply`, and so you should see the same result as setting it to a non-existing battery on a laptop.

**Reply by mobsenpai** [reply]
Tq
